utilities/utility.py

```diff
@@ -14,6 +16,35 @@ def update_entity_fields(entity, **kwargs):
     return entity
 
 
+def cascade_soft_delete(parent_model, child, parent_id):
+    @event.listens_for(parent_model, 'after_update')
+    def receive_after_update(mapper, connection, target):
+        listen_for_after_flush_event(target, child, parent_id)
+
+
+def listen_for_after_flush_event(target, child, parent_id):
+    @event.listens_for(db_session, "after_flush", once=True)
+    def receive_after_flush(session, context):
+        if target.state == 'archived':
+            child_model = getattr(api, child)
+            # Check for resource name is due to the improper
+            # naming of folder room_resource and class Resource
+            child_object = getattr(
+                child_model.models,
+                'Resource' if child.capitalize() == 'Room_resource'
+                else child.capitalize()
+            )
+            parent_s_id = getattr(child_object, parent_id)
```

Comment:
Did you mean `parents_id`? If not so then use a more descriptive variable naming